homeassistant/components/dio_chacon/config_flow.py

```diff
@@ -0,0 +1,72 @@
+"""Config flow for dio_chacon integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api import DIOChaconAPIClient
+from dio_chacon_wifi_api.exceptions import DIOChaconAPIError, DIOChaconInvalidAuthError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_UNIQUE_ID, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class DioChaconConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for dio_chacon."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                # Validate the user name and password and retrieve the technical user id.
+                client = DIOChaconAPIClient(
+                    user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                )
+                _user_id: str = await client.get_user_id()
+                await client.disconnect()
+            except DIOChaconAPIError:
+                errors["base"] = "cannot_connect"
+            except DIOChaconInvalidAuthError:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            else:
+                # Check if already configured
+                await self.async_set_unique_id(_user_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title="Dio Chacon " + user_input[CONF_USERNAME],
+                    data={
+                        CONF_USERNAME: user_input[CONF_USERNAME],
+                        CONF_PASSWORD: user_input[CONF_PASSWORD],
+                        CONF_UNIQUE_ID: _user_id,
```

Comment:
No need to store it as we can access it at `config_entry.unique_id`

Reply:
ok